ubi_manifest/worker/tasks/content_audit.py

```diff
@@ -215,19 +208,31 @@ def _latest_input_rpms(fts: list[Future[set[UbiUnit]]]) -> list[UbiUnit]:
 
 
 def _latest_input_mds(fts: list[Future[set[UbiUnit]]]) -> list[UbiUnit]:
-    out = []
+    latest_mds = []
+    module_map = defaultdict(list)
+
     for ft in as_completed(fts):
-        out.extend(list(ft.result()))
-    keep_n_latest_modules(out)
-    return out
+        for md in ft.result():
+            module_map[f"{md.name}:{md.stream}"].append(md)
+    for module_group in module_map.values():
+        keep_n_latest_modules(module_group)
+        latest_mds.extend(module_group)
+
+    return latest_mds
 
 
 def _latest_input_mdds(fts: list[Future[set[UbiUnit]]]) -> list[UbiUnit]:
```

Comment:
This probably is redundant after all, as there is always one `modulemd_defaults` unit for given module or module:stream. But I guess we can keep this here (but deciding on what's latest by profiles probably isn't really accurate and because of design - one mdd for one md we don't need to do this).
I would say that latest would the unit existing in the input repo and that's it.

Reply:
So, we don't need to check modulemd_defaults up-to-dateness at all?

If so, can we strike "modulemd_defaults" from Jira issue description?
> * It should check that all available content is up-to-date (rpm non-modular -+ modular-, modulemd, modulemd_defaults)

Reply:
Also, I would save removal of this stuff for another commit since it wasn't touched by this one.

Reply:
Well, we should check modulemd_defaults but the should only be based on `profiles` field (which already is), only thing that is redundant is the getting latest modulemd_default because of the design.

Reply:
I think you can add the fix in another commit in this PR - but it's up to you.

Reply:
Done!